pkg/cmd/adm/restart.go

```diff
@@ -1,157 +1,213 @@
+		runtimeclient.InNamespace(deployment.Namespace)); err != nil {
 		return err
 	}
-	deploymentList := "\n"
-	for _, deployment := range deployments.Items {
-		deploymentList += fmt.Sprintf("%s\n", deployment.Name)
+	ctx.Printlnf("Starting to delete the pods")
+	//delete pods
+	for _, pod := range pods.Items {
+		pod := pod // TODO We won't need it after upgrading to go 1.22: https://go.dev/blog/loopvar-preview
+		if err := cl.Delete(ctx, &pod); err != nil {
+			return err
+		}
 	}
-	term.PrintContextSeparatorWithBodyf(deploymentList, "Existing deployments in %s namespace", ns)
 	return nil
+
 }
 
-func scaleToZero(cl runtimeclient.Client, namespacedName types.NamespacedName) (int32, error) {
-	// get the deployment
-	deployment := &appsv1.Deployment{}
-	if err := cl.Get(context.TODO(), namespacedName, deployment); err != nil {
-		return 0, err
+func restartNonOperatorDeployments(ctx *clicontext.CommandContext, deployment appsv1.Deployment, f cmdutil.Factory, ioStreams genericclioptions.IOStreams) error {
+
+	o := kubectlrollout.NewRolloutRestartOptions(ioStreams)
+
+	if err := o.Complete(f, nil, []string{"deployment"}); err != nil {
+		panic(err)
 	}
-	// keep original number of replicas so we can bring it back
-	originalReplicas := *deployment.Spec.Replicas
-	zero := int32(0)
-	deployment.Spec.Replicas = &zero
 
-	// update the deployment so it scales to zero
-	return originalReplicas, cl.Update(context.TODO(), deployment)
+	o.Resources = []string{"deployment/" + deployment.Name}
+
+	if err := o.Validate(); err != nil {
+		panic(err)
+	}
+	ctx.Printlnf("Running the rollout restart command for non-Operator deployment %v", deployment.Name)
+	return o.RunRestart()
 }
 
-func scaleBack(term ioutils.Terminal, cl runtimeclient.Client, namespacedName types.NamespacedName, originalReplicas int32) error {
-	return wait.Poll(500*time.Millisecond, 10*time.Second, func() (done bool, err error) {
-		term.Println("")
-		term.Printlnf("Trying to scale the deployment back to '%d'", originalReplicas)
-		// get the updated
-		deployment := &appsv1.Deployment{}
-		if err := cl.Get(context.TODO(), namespacedName, deployment); err != nil {
-			return false, err
-		}
-		// check if the replicas number wasn't already reset by a controller
-		if *deployment.Spec.Replicas == originalReplicas {
-			return true, nil
-		}
-		// set the original
-		deployment.Spec.Replicas = &originalReplicas
-		// and update to scale back
-		if err := cl.Update(context.TODO(), deployment); err != nil {
-			term.Printlnf("error updating Deployment '%s': %s. Will retry again...", namespacedName.Name, err.Error())
-			return false, nil
-		}
-		return true, nil
-	})
+func checkRolloutStatus(ctx *clicontext.CommandContext, f cmdutil.Factory, ioStreams genericclioptions.IOStreams, labelSelector string) error {
+
+	cmd := kubectlrollout.NewRolloutStatusOptions(ioStreams)
+
+	if err := cmd.Complete(f, []string{"deployment"}); err != nil {
+		panic(err)
+	}
+	cmd.LabelSelector = labelSelector
```

Comment:
is there any reason for using the labelSelector instead of the deployment name?
in the standard deployment case you call it with 
```
checker(ctx, "toolchain.dev.openshift.com/provider=codeready-toolchain")
```
which would mean that it would check all deployments with this label with this label  (it can be more of them) for every call in the loop.

Reply:
there can be same deployment names, and we want to only check status of the correct one, also i do not see any parameter where in rollout status where , a name can be passed

Reply:
Yes but ,the valid args given for it are 
https://github.com/kubernetes/kubectl/blob/169a9524172716d4dbcceb8376698e5a630e578c/pkg/cmd/rollout/rollout_status.go#L99 which doesn't contain /name. , the same reason i added /deployment in rolloutstatus via resources https://github.com/kubesaw/ksctl/pull/79/files#diff-281b31e7723a650782511ee228ab82b4065dd935ce2d4e4c0fb0f491ff6732bfR172 ,
and somehow when i was tested that specific code locally in a cluster giving /name in complete didn't work, so switched to this .. but i might be wrong.

Reply:
ok, let me ask one simple question, when you are saying that it doesn't support providing a name of the deployment, how can it work in kubectl when executed on command line?

Reply:
Okay ,i get your point, l will try adding the deployment name in complete. and also keep the label selector

Reply:
updated it

Reply:
I'm confused, why do you want to keep the label selector? the deployment name should be enough

Reply:
i removed the label selector